ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -314,4 +306,22 @@ - (NSDictionary *)constantsToExport
     resolve(@(batteryLevel));
 }
 
+RCT_EXPORT_METHOD(getUserAgent:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (_webView == nil) {
+            _webView = [[WKWebView alloc] initWithFrame:CGRectZero];
+        }
+        [_webView evaluateJavaScript:@"navigator.userAgent" completionHandler:^(id result, NSError * error) {
+            if (error == nil) {
+                if (result != nil) {
+                    resolve([NSString stringWithFormat:@"%@", result]);
```

Comment:
is `result` not a string?

Reply:
Implicit cast can be dangerous, I'll add a Type check